### ISSUE TITLE
currentBranch logic seems broken

### ISSUE THREAD

**Issue by pantovad** [ilovemilk/semver-git-plugin#23]
When building from release/.* or hotfix/.* branches (for example release/5.3.0 or hotfix/5.3.1) showInfo fails to return branch details and thus branch specific configuration not usable:

From release/5.3.0:
> Task :showInfo
Branch name: 0
Branch group: 0
Branch id: 0
Commit: 17bdbf690ea59b3673c845bbc430d2edd121a9e4
Short commit: 17bdbf6
Tag: none
Last tag: 5.2.1-4-g17bdbf6
Dirty: false
Version: 5.3.0-BUILD.31.sha.17bdbf6-SNAPSHOT
Version major: 5
Version minor: 3
Version patch: 0
Version pre release: none
Version build: none

From hotfix/5.3.1:
> Task :showInfo
Branch name: 1
Branch group: 1
Branch id: 1
Commit: 995e021d69052973e4b9f2b2fd6c3f3f25e7dce6
Short commit: 995e021
Tag: none
Last tag: 5.2.1-1-g995e021
Dirty: false
Version: 5.3.0-BUILD.28.sha.995e021-SNAPSHOT
Version major: 5
Version minor: 3
Version patch: 0
Version pre release: none
Version build: none

Configuration:
```
semver {
    snapshotSuffix = "SNAPSHOT"
    dirtyMarker = "dirty"
    branches {
        branch {
            regex = "develop"
            incrementer = "MINOR_INCREMENTER"
            formatter = { "${semver.info.version.major}.${semver.info.version.minor}.${semver.info.version.patch}-DEV.${semver.info.count}.sha.${semver.info.shortCommit}" }
        }
        branch {
            regex = "release/.+"
            incrementer = "MINOR_INCREMENTER"
            formatter = { "${semver.info.version.major}.${semver.info.version.minor}.${semver.info.version.patch}-RC.${semver.info.count}.sha.${semver.info.shortCommit}" }
        }
        branch {
            regex = "hotfix/.+"
            incrementer = "PATCH_INCREMENTER"
            formatter = { "${semver.info.version.major}.${semver.info.version.minor}.${semver.info.version.patch}-HOTFIX.${semver.info.count}.sha.${semver.info.shortCommit}" }
        }
        branch {
            regex = ".+"
            incrementer = "MINOR_INCREMENTER"
            formatter = { "${semver.info.version.major}.${semver.info.version.minor}.${semver.info.version.patch}-BUILD.${semver.info.count}.sha.${semver.info.shortCommit}" }
        }
    }
}
```

**Comment by ilovemilk** [comment]
This should be fixed with version 2.2.1!

Thanks for reporting this bug.

**Comment by pantovad** [comment]
You are a star Matthias ! Thx vm for the quick response !
